tree/ntuple/v7/test/CMakeLists.txt

```diff
@@ -36,6 +36,10 @@ ROOT_ADD_GTEST(ntuple_serialize ntuple_serialize.cxx LIBRARIES ROOTDataFrame ROO
 ROOT_ADD_GTEST(ntuple_types ntuple_types.cxx LIBRARIES ROOTDataFrame ROOTNTuple MathCore CustomStruct)
 ROOT_ADD_GTEST(ntuple_view ntuple_view.cxx LIBRARIES ROOTDataFrame ROOTNTuple MathCore CustomStruct)
 ROOT_ADD_GTEST(ntuple_zip ntuple_zip.cxx LIBRARIES ROOTDataFrame ROOTNTuple MathCore CustomStruct)
+# Some of these tests require cxxmodules support, e.g. for typedef resolution.
```

Comment:
What happens in the non-cxxmodules case? I.e. what is the typedef resolution failure you're referring to here?

Reply:
In `RFieldBase::Create(const std::string &fieldName, const std::string &typeName)`, we normalize the `typeName`, which includes resolving typedefs to one of the other types known to RNTuple.  We do that via `TClassEdit::ResolveTypedef()`.

However, in the non-cxxmodules case, information for the typedef is not available to cling, which causes `TClassEdit::ResolveTypedef()` to return the input untouched.  This in turn, causes RNTuple to throw an exception:
```
Field m_sgkey has unknown type SG::sgkey_t (unchecked RResult access!)
```

Of course we could do `gInterpreter->ProcessLine("#include \"foo.h\"")`, but given that this test is based on a use case  by Marcin Nowak (EP-UAT), I think it should be closer to what a user would do/expect.

Reply:
> I think it should be closer to what a user would do

That depends. These are unit tests, not integration tests, and as such they should probably have an easy, self-contained setup - in contrast to an experiment's use case. We have tests that assert that the injection of reflection data works, that libraries are loaded if data is needed etc. Here, failure should be mostly limited to functionality in RNTuple, and we should provide a concise test that allow us to reduce the debugging complexity. Makes sense?

Reply:
Make sense.  Thanks!